### PR TITLE
Update Tools and API Versions

### PR DESCRIPTION
**Pull request by razo7**
Bump FAR to the latest tool and API versions (e.g. K8s, OCP, Controller-gen, Kustomize, and Operator-SDK).

In addition it also includes more stuff:

- Fix for writing/deleting the `envtest` target directory
- Fetching the machine client (for E2E) using OpenShift client-go

[ECOPROJECT-1403](https://issues.redhat.com//browse/ECOPROJECT-1403)

### CODE REVIEW

**Comment by razo7** [comment]
Golang must be v1.20 prior to merging this PR. 
/hold
Wait for #57 to be merged

**Comment by razo7** [comment]
E2E test will fail as it needs https://github.com/openshift/release/pull/40248 to be merged, so Golang version on E2E will be updated

**Comment by mshitrit** [comment]
/lgtm
/hold
Giving others chance to review

**Comment by clobrano** [comment]
/lgtm

Giving a chance to get more feedbacks, feel free to unhold
/hold

**Comment by razo7** [comment]
I hope #20 will be merged soon, and I prefer to wait for this PR to be merged. Since this PR introduce huge bump (by time of versions, not necessarily it's true for functionality of FAR), thus I would prefer to wait for #20.

**Comment by razo7** [comment]
E2E PR was merged and this PR is ready for review

File: bundle/manifests/fence-agents-remediation.medik8s.io_fenceagentsremediations.yaml [hunk 1]
```diff
@@ -73,5 +73,5 @@ status:
   acceptedNames:
     kind: ""
     plural: ""
-  conditions: []
-  storedVersions: []
+  conditions: null
```
Comment (slintes) [inline bundle/manifests/fence-agents-remediation.medik8s.io_fenceagentsremediations.yaml:76]: interesting... in /config this status is removed completely from the CRDs, here just the list values change :thinking: :shrug:

File: config/crd/kustomization.yaml [hunk 2]
```diff
@@ -6,7 +6,6 @@ resources:
 - bases/fence-agents-remediation.medik8s.io_fenceagentsremediationtemplates.yaml
 #+kubebuilder:scaffold:crdkustomizeresource
 
-patchesStrategicMerge:
```
Comment (slintes) [inline config/crd/kustomization.yaml:9]: so... how would we enable webhooks now?

File: config/default/kustomization.yaml [hunk 3]
```diff
@@ -45,31 +37,15 @@ patchesStrategicMerge:
 #- webhookcainjection_patch.yaml
 
 # the following config is for teaching kustomize how to do var substitution
-vars:
-# [CERTMANAGER] To enable cert-manager, uncomment all sections with 'CERTMANAGER' prefix.
```
Comment (slintes) [inline config/default/kustomization.yaml:49]: not sure if it's a good idea to remove this certmanager config...

**Review by slintes — CHANGES_REQUESTED** [review]
there changes not relevant to the PR title in the PR
- please revert the kustomize changes and put them in into their own PR. And when doing so, either delete all unneeded and/or commented configs, or migrate all of them to the new format. What we have in this PR now is a mix of everything. I'd prefer to migrate all (and keep them commented), it will make enabling webhooks later on much easier
- there also code changes, probably also worth their own PR?

**Comment by razo7** [comment]
Ok, I will create PR for changing the OCP client-go and fetching machine client in E2E test and in another PR I will update and fix deprecation warnings due to Kustomize v5 changes.
> migrate all of them to the new format.

When you say new format are you referring to Kubebuilder `go/v3` to `g0/v4`?  I remembered that go/v4 was still as alpha, and looking again I see that it has already been stabilized ECOPROJECT-1315

**Comment by razo7** [comment]
> there are also code changes, probably also worth their own PR?

I was thinking of doing another PR for them.

>Ok, I will create PR for changing the OCP client-go and fetching machine client in E2E test

But looking again on the changes, they are quite minor IMHO, and due to bumping the K8s and OCP APIs I was aware of an error with `vet` under vendor due to the old `github.com/openshift/machine-api-operator` package.
Therefore, I would prefer to let the these code changes remain in this PR. 
I can update the PR's title if you still see a need for that.
WDYT?

**Comment by slintes** [comment]
> > there are also code changes, probably also worth their own PR?
> 
> I was thinking of doing another PR for them.
> 
> > Ok, I will create PR for changing the OCP client-go and fetching machine client in E2E test
> 
> But looking again on the changes, they are quite minor IMHO, and due to bumping the K8s and OCP APIs I was aware of an error with `vet` under vendor due to the old `github.com/openshift/machine-api-operator` package. Therefore, I would prefer to let the these code changes remain in this PR. I can update the PR's title if you still see a need for that. WDYT?

ack, wfm

**Comment by razo7** [comment]
Updating to kustomize v5, and removing new deprecation warnings will be done in a follow up PR when we change the project layout to go/v4 - [ECOPROJECT-1315](https://issues.redhat.com//browse/ECOPROJECT-1315)

**Review by slintes — APPROVED** [review]

**Pull request merged**